zenoh-flow/src/runtime/dataflow/instance/mod.rs

```diff
@@ -354,6 +354,19 @@ impl DataflowInstance {
         Ok(manager.await?)
     }
 
+    /// Finalized the given node.
+    /// Finalizing a node means cleaning up its state.
+    ///
+    /// # Errors
+    /// If fails if the node is not found.
+    pub async fn clean_node(&mut self, node_id: &NodeId) -> ZFResult<()> {
```

Comment:
Do we need to return an error here? If not, maybe we can log there that the node we are trying to clean does not exist which would simplify the code in the daemon.

What do you think?

Reply:
I prefer returning the error here and then leaving the caller to decide if just log it or not.
The idea behind this is that the instance does not know the action to be taken, so it reports any error to the caller that should have the intelligence to deal with it.